pkg/log/zap/zap_test.go

```diff
@@ -296,3 +300,128 @@ var _ = Describe("Zap logger setup", func() {
 		})
 	})
 })
+
+var _ = Describe("Zap flag options setup", func() {
+	fromFlags := Options{}
+
+	BeforeEach(func() {
+		flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError) //flags are now reset
+	})
+
+	Context("with -zap-devel options provided", func() {
+		It("Should set dev=true with relative options.", func() {
+			os.Args = []string{"cmd", "-zap-devel=true"}
+			fromFlags.BindFlags(flag.CommandLine)
+			flag.Parse()
+			out := Options{}
+			UseFlagOptions(&fromFlags)(&out)
+			out.addDefaults()
+
+			Expect(out.Development).To(BeTrue())
+			Expect(out.Level.Enabled(zapcore.DebugLevel)).To(BeTrue())
+			Expect(out.StacktraceLevel.Enabled(zapcore.WarnLevel)).To(BeTrue())
+			cfg := zap.NewDevelopmentEncoderConfig()
+			Expect(out.Encoder).Should(BeAssignableToTypeOf(zapcore.NewConsoleEncoder(cfg)))
+			Expect(out.EncoderConfigOptions).To(BeNil())
+		})
+		It("Should set dev=false with relative options.", func() {
+			os.Args = []string{"cmd", "-zap-devel=false"}
+			fromFlags.BindFlags(flag.CommandLine)
+			flag.Parse()
+			out := Options{}
+			UseFlagOptions(&fromFlags)(&out)
+			out.addDefaults()
+
+			Expect(out.Development).To(BeFalse())
+			Expect(out.Level.Enabled(zapcore.InfoLevel)).To(BeTrue())
+			Expect(out.StacktraceLevel.Enabled(zapcore.ErrorLevel)).To(BeTrue())
+			cfg := zap.NewProductionEncoderConfig()
+			Expect(out.Encoder).Should(BeAssignableToTypeOf(zapcore.NewJSONEncoder(cfg)))
+			Expect(out.EncoderConfigOptions).To(BeNil())
+		})
+	})
+
+	Context("with -zap-Encoder and -zap-Time-Encoder options provided.", func() {
+
+		It("Should set JSON Encoder, with given TimEncoder options.", func() {
+			os.Args = []string{"cmd", "-zap-encoder=json", "-zap-time-encoder=millis"}
+			fromFlags.BindFlags(flag.CommandLine)
+			flag.Parse()
+			out := Options{}
+			UseFlagOptions(&fromFlags)(&out)
+			out.addDefaults()
+
+			// Set logger with custom io.writer
+			logOut := new(bytes.Buffer)
+			sink := zapcore.AddSync(logOut)
+			log := zap.New(zapcore.NewCore(&KubeAwareEncoder{Encoder: out.Encoder, Verbose: out.Development}, sink, out.Level))
```

Comment:
Nit: instead of the above, would the following work?

```suggestion
			logOut := new(bytes.Buffer)
			log := New(UseFlagOptions(&fromFlags), WriteTo(logOut))
```

In this cases, I don't think we would even need the `out` variable.

Ditto elsewhere.